packages/camera/camera_windows/windows/camera_plugin.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef PACKAGES_CAMERA_CAMERA_WINDOWS_WINDOWS_CAMERA_PLUGIN_H_
+#define PACKAGES_CAMERA_CAMERA_WINDOWS_WINDOWS_CAMERA_PLUGIN_H_
+
+#include <flutter/flutter_view.h>
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar_windows.h>
+#include <flutter/standard_method_codec.h>
+
+#include <functional>
+
+#include "camera.h"
+#include "capture_controller.h"
+#include "capture_controller_listener.h"
+
+namespace camera_windows {
+using flutter::MethodResult;
+
+class CameraPlugin : public flutter::Plugin,
+                     public VideoCaptureDeviceEnumerator {
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrarWindows *registrar);
+
+  CameraPlugin(flutter::TextureRegistrar *texture_registrar,
+               flutter::BinaryMessenger *messenger);
+
+  // Creates a plugin instance with the given CameraFactory instance.
+  // Exists for unit testing with mock implementations.
+  CameraPlugin(flutter::TextureRegistrar *texture_registrar,
+               flutter::BinaryMessenger *messenger,
+               std::unique_ptr<CameraFactory> camera_factory);
+
+  virtual ~CameraPlugin();
+
+  // Disallow copy and move.
+  CameraPlugin(const CameraPlugin &) = delete;
+  CameraPlugin &operator=(const CameraPlugin &) = delete;
+
+  // Called when a method is called on plugin channel;
+  void HandleMethodCall(const flutter::MethodCall<> &method_call,
+                        std::unique_ptr<MethodResult<>> result);
+
+ protected:
```

Comment:
Why are there protected elements? Plugins aren't meant to be subclassed.

Reply:
The standard way to access internals—when it is absolutely necessary—in C++ tests is `friend` annotations granting access to a test fixture.